src/Sulu/Bundle/AdminBundle/Resources/js/containers/Form/fields/Select.js

```diff
@@ -0,0 +1,86 @@
+// @flow
+import React from 'react';
+import MultiSelectComponent from '../../../components/MultiSelect';
+import type {FieldTypeProps} from '../../../types';
+
+const MISSING_VALUES_OPTIONS = 'The "values" option has to be set for the SingleSelect FieldType';
+
+type Props = FieldTypeProps<Array<string | number>>;
+
+export default class Select extends React.Component<Props> {
+    constructor(props: FieldTypeProps<Array<string | number>>) {
+        super(props);
+
+        const {onChange, schemaOptions, value} = this.props;
+
+        if (!schemaOptions) {
+            return;
+        }
+
+        let {
+            default_values: {
+                value: defaultValues,
+            } = {},
+        } = schemaOptions;
+
+        if (defaultValues === undefined || defaultValues === null) {
+            return;
+        }
+
+        if (!Array.isArray(defaultValues)) {
+            throw new Error('The "default_values" schema option must be an array!');
+        }
+
+        defaultValues = defaultValues.map(({value, name}) => {
+            const defaultValue = value === undefined || value === null ? name : value;
+
+            if (typeof defaultValue !== 'number' && typeof defaultValue !== 'string') {
+                throw new Error('A single schema option of "default_values" must be a string or number');
+            }
+
+            return defaultValue;
+        });
+
+        if (value === undefined) {
+            onChange(defaultValues);
+        }
+    }
+
+    handleChange = (value: Array<string | number>) => {
+        const {onChange, onFinish} = this.props;
+
+        onChange(value);
+        onFinish();
+    };
+
+    render() {
+        const {schemaOptions, disabled, value} = this.props;
+        if (!schemaOptions) {
+            throw new Error(MISSING_VALUES_OPTIONS);
+        }
+
+        const {values} = schemaOptions;
+
+        if (!Array.isArray(values.value)) {
+            throw new Error(MISSING_VALUES_OPTIONS);
+        }
+
+        return (
+            <MultiSelectComponent disabled={!!disabled} onChange={this.handleChange} values={value || []}>
+                {values.value.map(({name, value, title}) => {
+                    value = value === undefined || value === null ? name : value;
```

Comment:
Didn't we agree on always using the name? :thinking:

Reply:
@danrot Yeah but there is one disadvantage/Problem with the name. In PHP a int to string cast is done on setName. So the name cannot be a number. 🙈

Reply:
https://github.com/sulu/sulu/blob/e2f26cf7a0b4a6bae52da0bfda56c0454973e708/src/Sulu/Component/Content/Metadata/ItemMetadata.php#L111

Reply:
https://github.com/sulu/sulu/blob/e2f26cf7a0b4a6bae52da0bfda56c0454973e708/src/Sulu/Bundle/AdminBundle/Resources/js/types.js#L17

Reply:
/cc @chirimoya